<?xml version="1.0"?>
<launch>
    <arg name="robot" default="$(find amr_ros)/config"/>

    <!--start-->

    <node pkg="joy" name="joy_node" type="joy_node">
	    <param name="device" value="/dev/input/js0"/>
    </node>
    <node pkg="manual_controller" name="manual" type="manual" output="screen"/>

    <!--end-->

    <node pkg="ros_whill" type="ros_whill" name="whill" output="screen" >

	    <remap from="/whill/controller/cmd_vel" to="/cmd_vel" />
	    <remap from="/whill/pose_particle_localizer" to="/pose_particle_localizer" />
	    <remap from="/whill/pose_odom" to="/pose_odom" />
	<!-- Initial Speed Profile -->
	<rosparam file="$(find ros_whill)/params/initial_speedprofile.yaml" command="load" />
	<!-- Keep Connected: If you specify true,ros_whill try to keep connection by re-opening port and sending power on command.
	    If you'd like to turn WHILL off by command or manually while using ros_whill, set value to false.
	    Otherwise WHILL automatically turns on.
	-->
	<param name="keep_connected" value="true" />
	<!-- Serial Device name -->
	<param name="serialport" value="/dev/ttyUSB0"/>
	<!-- Output Odometry tf or not. If other node publishes odom->base_link tf, Change value to "false" -->
	<param name="publish_tf" value="true"/>
    </node>

    <include file="$(find rplidar_ros)/launch/rplidar_a3.launch">
	<arg name="cut_angle"     value="true"/>
	<arg name="left_degrees"  value="-90"/>
    </include>

    <include file="$(find rt_usb_9axisimu_driver)/launch/rt_usb_9axisimu_driver.launch" >

    </include>

</launch>
